cdap-examples/SpamClassifier/src/test/java/co/cask/cdap/examples/sparkstreaming/SpamClassifierTest.java

```diff
@@ -0,0 +1,186 @@
+  public void test() throws Exception {
+
+    // Deploy the KafkaIngestionApp application
+    ApplicationManager appManager = deployApplication(SpamClassifier.class);
+
+    ingestTrainingData();
+
+    publishKafkaMessages();
+
+    // start spark streaming program
+    SparkManager sparkManager = appManager.getSparkManager(SpamClassifierProgram.class.getSimpleName());
+    Map<String, String> runtimeArgs = new HashMap<>();
+    runtimeArgs.put("kafka.brokers", "127.0.0.1:" + kafkaPort);
+    runtimeArgs.put("kafka.topics", KAFKA_TOPIC);
+    sparkManager.start(runtimeArgs);
+
+    // Start and wait for service to start
+    final ServiceManager serviceManager = appManager.getServiceManager(SpamClassifier.SERVICE_HANDLER).start();
+    serviceManager.waitForStatus(true);
+
+    // wait for spark streaming program to write to dataset
+    Tasks.waitFor(HttpURLConnection.HTTP_OK, new Callable<Integer>() {
+      @Override
+      public Integer call() throws Exception {
+        return getResponse(serviceManager, "1").getResponseCode();
+      }
+    }, 60, TimeUnit.SECONDS);
+
+    // stop spark program
+    sparkManager.stop();
+    sparkManager.waitForFinish(10, TimeUnit.SECONDS);
+
+    // Request total pages for a page rank and verify it
+    testClassification(serviceManager, "1", SpamClassifier.SpamClassifierServiceHandler.SPAM);
+    testClassification(serviceManager, "2", SpamClassifier.SpamClassifierServiceHandler.HAM);
+
+    appManager.stopAll();
+  }
+
+  private void testClassification(ServiceManager serviceManager, String messageId, String expected) throws IOException {
+    HttpResponse response = getResponse(serviceManager, messageId);
+    Assert.assertEquals(HttpURLConnection.HTTP_OK, response.getResponseCode());
+    Assert.assertEquals(expected, response.getResponseBodyAsString());
+  }
+
+  private HttpResponse getResponse(ServiceManager serviceManager, String messageId) throws IOException {
+    URL url = new URL(serviceManager.getServiceURL(15, TimeUnit.SECONDS),
+                      SpamClassifier.SpamClassifierServiceHandler.CLASSIFICATION_PATH + "/" + messageId);
+    return HttpRequests.execute(HttpRequest.get(url).build());
+  }
+
+  private void publishKafkaMessages() {
+    KafkaPublisher publisher = kafkaClient.getPublisher(KafkaPublisher.Ack.ALL_RECEIVED, Compression.NONE);
+    KafkaPublisher.Preparer preparer = publisher.prepare(KAFKA_TOPIC);
+
+    preparer.add(Charsets.UTF_8.encode("1:REMINDER FROM O2: To get 2.50 pounds free call credit and details of great " +
+                                         "offers pls reply 2 this text with your valid name, house no and postcode"),
+                 "1"); // spam
+    preparer.add(Charsets.UTF_8.encode("2:I will call you later"), "2"); // ham
+    preparer.send();
+  }
+
+  private void ingestTrainingData() throws IOException {
+    StreamManager streamManager = getStreamManager(SpamClassifier.STREAM);
+
+    try (BufferedReader reader = new BufferedReader(
```

Comment:
You can just use the `streamManager.send(File)` method instead of doing a custom loop.

```java
streamManager.send(new File(Resources.getResource("trainingData.txt").toURI()), "text/plain");
```